akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -175,27 +188,84 @@ private[akka] class Shard(typeName: String,
     None
   }
 
+  val lease = settings.leaseSettings.map(
+    ls ⇒
+      LeaseProvider(context.system).getLease(s"${context.system.name}-shard-$typeName-$shardId",
```

Comment:
that should be unique, but if the singleton naming is changed to use a path this should use the same convention

Reply:
One con of this is that the user decides the cluster singleton manager path but the shard path is internal decision so maybe this name is nicer?